bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-persistence.js

```diff
@@ -200,20 +233,61 @@ export default function defineOHBlocks_Persistence (f7, isGraalJs) {
   */
   javascriptGenerator['oh_get_persistence_lastupdate'] = function (block) {
     const itemName = javascriptGenerator.valueToCode(block, 'itemName', javascriptGenerator.ORDER_ATOMIC)
+    const inputType = block.getInput('itemName').connection.targetBlock().outputConnection.getCheck()[0]
+    const persistenceName = javascriptGenerator.valueToCode(block, 'persistenceName', javascriptGenerator.ORDER_NONE)
+    const persistenceExtension = (persistenceName === '\'default\'') ? '' : ((!isGraalJs) ? ',' : '') + ` ${persistenceName}`
+
+    let itemCode = toItemCode(itemName, inputType)
 
     if (isGraalJs) {
-      return [`items.getItem(${itemName}).history.lastUpdate()`, 0]
+      return [`${itemCode}.history.lastUpdate(${persistenceExtension})`, 0]
     } else {
       const { dtf, zdt, getZonedDatetime } = addDateSupport()
       const persistence = addPersistence()
-      let code = `${persistence}.lastUpdate(itemRegistry.getItem(${itemName}))`
+      let code = `${persistence}.lastUpdate(${itemCode}${persistenceExtension})`
       return [code, 0]
     }
   }
 
+  function toItemCode (itemName, inputType) {
```

Comment:
Would it make sense to put this into the `utils.js` because there are other blocks that can use this as well?
And I'd rename it to `generateItemCode`.

Reply:
I will eventually if in the future it would be used somewhere else but until then I would recommend to keep it here. I changed the name.

Reply:
The functionality provided by this function is also used in the eventbus blocks and probably others, would be nice if you could provide a follow-up PR to refactor those to use this function.